docs/index.md

```diff
@@ -12,13 +12,29 @@ This repository contains a Terraform provider for managing resources in a [Mater
 Configure the provider by adding the following block to your Terraform project:
 
 ```terraform
-# Configuration-based authentication
+# =============================================================================
+# Materialize Cloud (SaaS) Configuration
+# =============================================================================
+# Use this configuration for Materialize Cloud environments.
+# This provides access to ALL provider resources including:
+# - App passwords, users, SSO, SCIM resources
+# - All database resources (clusters, sources, sinks, etc.)
+#
 provider "materialize" {
   password       = var.materialize_password # optionally use MZ_PASSWORD env var
-  default_region = "aws/us-east-1"          # optionally use MZ_REGION env var
+  default_region = "aws/us-east-1"          # optionally use MZ_DEFAULT_REGION env var
 }
 
-# Self-hosted Materialize authentication
+# =============================================================================
+# Self-Hosted Materialize Configuration
+# =============================================================================
+# Use this configuration for self-hosted Materialize instances.
+# 
+# ⚠️  IMPORTANT LIMITATIONS:
+# - Frontegg-dependent resources are NOT available (app passwords, users, SSO, SCIM)
+# - Only database resources are available (clusters, sources, sinks, schemas, etc.)
+# - No organization or identity management features
```

Comment:
Interesting... do we support `create role ... with password, superuser...` in this terraform? That might be a viable callout here for identity management. If we don't, that might be a reasonable feature to add in the future.